### PR TITLE
Allow to select channels to mirror

### LINKED ISSUE #5: Allow filtering by channel
To minimize the amount of downloads for minimal mirror, it would be nice to be able to mirror only specific channel. With combination with above version, this should allow downloading only one/few images.

### PR DESCRIPTION
**Pull request by invidian**
So if someone wants to mirro only single version of Flatcar, this makes
it possible.

Closes #5.

Signed-off-by: Mateusz Gozdek <mateusz@kinvolk.io>

### CODE REVIEW

File: flatcar-release-mirror.sh [hunk 1]
```diff
@@ -150,13 +151,15 @@ while [[ "$#" -gt 0 ]]; do case $1 in
...
              echo "--logfile FILE             Write detailed log to FILE (can also be /dev/stderr)"
+             echo "--channels CHANNELS        Coma-separated list of channels to mirror (e.g. stable,beta). By default all channels are mirrored."
              echo "--help                     Show flags"
              exit 0;;
   --logfile) shift; [[ "$#" -eq 0 ]] && (echo "Expecting value after flag"; exit 1); logfile=$(readlink -f "$1")
              [[ -z "$logfile" ]] && (echo "Error: readlink utility missing?" >> /dev/stderr ; exit 1);;
   --not-files) shift; [[ "$#" -eq 0 ]] && (echo "Expecting value after flag"; exit 1); not_files="$1";;
   --only-files) shift; [[ "$#" -eq 0 ]] && (echo "Expecting value after flag"; exit 1); only_files="$1";;
   --above-version) shift; [[ "$#" -eq 0 ]] && (echo "Expecting value after flag"; exit 1); above_version="$1";;
+  --channels) shift; [[ "$#" -eq 0 ]] && (echo "Expecting value after flat"; exit 1); channels="$1";;
```
Comment (pothos) [inline flatcar-release-mirror.sh:162]: Should be `flag` not `flat`

File: flatcar-release-mirror.sh [hunk 2]
```diff
@@ -171,7 +174,7 @@ else
   rm "/tmp/mirror-err" 2> /dev/null || true
 fi
 
-channels="stable beta alpha edge"
+channels=$(echo $channels | sed 's/,/ /g')
```
Comment (pothos) [inline flatcar-release-mirror.sh:177]: Please add quotes around `$()`

Comment (invidian) [reply flatcar-release-mirror.sh]: Fixed, thanks.

File: flatcar-release-mirror.sh (hunk 1 above)
Comment (invidian) [reply flatcar-release-mirror.sh]: Fixed, thanks.

**Comment by pothos** [comment]
By the way: For testing I use `--only-files contents`.

**Comment by pothos** [comment]
Ah, please update the README!

**Comment by invidian** [comment]
Done.

File: flatcar-release-mirror.sh [hunk 3]
```diff
@@ -55,14 +55,13 @@ download_file() {
 
   if [[ -f "$file" ]]; then
     local etag=$(caddy_etag "$file")
-    log=$(curl "$url" -I -H "If-None-Match: $etag" $CURLARGS 2>&1)
+    response_code=$(curl -s -o /dev/null -w "%{http_code}" "$url" -I -H "If-None-Match: $etag" $CURLARGS 2>&1)
```
Comment (pothos) [inline flatcar-release-mirror.sh:58]: Please make this a local variable

Comment (invidian) [reply flatcar-release-mirror.sh:59]: Done.

**Pull request merged**